lxd/storage/s3/miniod/miniod.go

```diff
@@ -127,32 +127,41 @@ func (p *Process) Stop(ctx context.Context) error {
 }
 
 // WaitReady waits until process is ready.
-func (p *Process) WaitReady(ctx context.Context) error {
-	adminClient, err := p.AdminClient()
-	if err != nil {
-		p.cancel.Cancel()
-		return err
-	}
+func (p *Process) WaitReady(ctx context.Context, timeout time.Duration) error {
```

Comment:
The timeout should be part of the ctx passed in right, rather than a separate argument?